### DISCUSSION TITLE
Custom pages folder

### DISCUSSION THREAD

**Discussion by k0mar12** [nuxt/nuxt#16670] [Questions] [answered]
Is it possible to add custom folder for pages ?
I mean, i need one default and one custom folders.

I know one way, i need to catch hook `pages:extend` and then pass there route that i need, but is it possible to push folder to scan as routes?

For example:

```
nuxt.hook('pages:extend', (pages) => {
    pages.pushDir('~/custom/dir')
})
```

Thanks

**Comment by k0mar12** [comment]
Im hust trying to implement ddd.

**Comment by k0mar12** [comment]
I have found this method:

```
export async function resolvePagesRoutes (): Promise<NuxtPage[]> {
  const nuxt = useNuxt()

  const pagesDirs = nuxt.options._layers.map(
    layer => resolve(layer.config.srcDir, layer.config.dir?.pages || 'pages')
  )

  const allRoutes = (await Promise.all(
    pagesDirs.map(async (dir) => {
      const files = await resolveFiles(dir, `**/*{${nuxt.options.extensions.join(',')}}`)
      // Sort to make sure parent are listed first
      files.sort()
      return generateRoutesFromFiles(files, dir)
    })
  )).flat()

  return uniqueBy(allRoutes, 'path')
}
```

As i can see, this method resolving default page folder. But dirs incoming from configs

**Comment by TheAlexLichter** [comment, marked as answer]
I think [extending the config via `extends`](https://github.com/nuxt/framework/tree/main/examples/advanced/config-extends) could help you there

**Reply by k0mar12** [reply]
This is exactly what i had been looking for. Thanks :)
